site/content/en/docs/Integration Patterns/canary-testing.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "Canary Testing a new Fleet"
+linkTitle: "Canary Testing"
+date: 2021-07-27T23:49:14Z 
+weight: 30 
+description: >
+  Run a small Fleet for the new version of your `GameServer` to ensure it works correctly, before rolling it out to all
+  your players.
+---
+
+To [canary release/test](https://martinfowler.com/bliki/CanaryRelease.html) a new `Fleet`, 
+we can run a small, fixed size `Fleet` of the new version of our GameServer, while also running the current stable 
```

Comment:
Based on this first paragraph, I was expecting the pattern to be something like:

- Always run 2 fleets, canary and prod
- Canary is fixed size, never changing
- Canary gets a new version, some players get that version
- Once canary is deemed "good", prod gets updated to the version in the canary fleet
- Repeat by updating canary to a newer version

I think my confusion is because you start out talking about a fixed size fleet here. What the pattern you describe does is more along the lines of "create a second fleet at the new version with a small size, verify it's ok, and then increase the size while decreasing the size of the first fleet"

This is akin to doing a rolling update on your own instead of letting the Agones controller do it for you.

Reply:
Given that this is contrast to doing fleet updates in place, I'd also like to see a link to https://agones.dev/site/docs/guides/fleet-updates/#rolling-update-strategy and a discussion of when to use this pattern vs. that pattern.

Reply:
Excellent point, I think my intent was aligned with yours, I just used language that was not clear:

> Over time, if monitoring those playing on the canary `Fleet` is working as expected, the size of the canary `Fleet` can be 
grown until you feel confident that it can replace the existing stable, production `Fleet`.   

When I thought "replace" I had a mental model of saying "do a rolling update to your stable Fleet" - I'll make the change to the language to reflect this.

>  I'd also like to see a link to https://agones.dev/site/docs/guides/fleet-updates/#rolling-update-strategy and a discussion of when to use this pattern vs. that pattern.

This is an excellent suggestion for a 'what next' link. I'll add this as well.